src/network/core/udp.cpp

```diff
@@ -99,7 +99,9 @@ void NetworkUDPSocketHandler::SendPacket(Packet *p, NetworkAddress *recv, bool a
 		}
 
 		/* Send the buffer */
-		int res = sendto(s.second, (const char*)p->buffer, p->size, 0, (const struct sockaddr *)send.GetAddress(), send.GetAddressLength());
+		ssize_t res = p->TransferOut([&s, &send](const char *buffer, size_t amount) {
+			return sendto(s.second, buffer, static_cast<int>(amount), 0, (const struct sockaddr *)send.GetAddress(), send.GetAddressLength());
```

Comment:
It is passed by reference and the address remains alive longer than the TransferOut function works, so that should not be a problem. The resolving should be fine. The lambdas are, as far as I know, not much more than glorified function pointers that C also knows.